[V2] docusaurus build becomes idle instead of throwing an error

## 🐛 Bug Report

`docusaurus build` doesn't throw an error on a failing build which results in high idle time.

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## To Reproduce

(Write your steps here:)

1. Clone the repo https://github.com/monapasan/docusaurus-idle-when-error
2. run `npm run build`


## Expected behavior

I would expect an error to be thrown and build to be finished.
<!--
  How did you expect your project to behave?
  It’s fine if you’re not sure your understanding is correct.
  Write down what you thought would happen.
-->


## Actual Behavior
You will see that docusaurus can't resolve a module in mdx file:
```
./docs/mdx.md
Module not found: Can't resolve '../../build/components' 
```
This then results in idle time
<!--
  Did something go wrong?
  Is something broken, or not behaving as you expected?
  Describe this section in detail, and attach screenshots if possible.
  Don't only say "it doesn't work"!
-->

## Reproducible Demo

https://github.com/monapasan/docusaurus-idle-when-error

<!--
  What happens if you skip this step?

  Someone will read your bug report, and maybe will be able to help you,
  but it’s unlikely that it will get much attention from the team. Eventually,
  the issue will likely get closed in favor of issues that have reproducible demos.

  Please remember that:

    * Issues without reproducible demos have a very low priority.
    * The person fixing the bug would have to do that anyway. Please be respectful of their time.
    * You might figure out the issues yourself as you work on extracting it.

  Thanks for helping us help you!
-->

--- a/packages/docusaurus/src/webpack/client.ts
+++ b/packages/docusaurus/src/webpack/client.ts
@@ -4,6 +4,8 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  */
+
+import chalk = require('chalk');
 import path from 'path';
 import {Configuration} from 'webpack';
 import merge from 'webpack-merge';
@@ -30,6 +32,26 @@ export function createClientConfig(props: Props): Configuration {
       runtimeChunk: true,
     },
     plugins: [
+      // Plugin to force terminate building if errors happened in the client bundle
+      {
+        apply: compiler => {
+          compiler.hooks.done.tap('client:done', stats => {
+            if (stats.hasErrors()) {
+              console.log(
+                chalk.red(
+                  'Client bundle compiled with errors therefore further build is impossible.',
+                ),
+              );
+
+              stats.toJson('errors-only').errors.forEach(e => {
+                console.error(e);
+              });
+
+              process.exit(1);
+            }
+          });
+        },
+      },
       new ChunkAssetPlugin(),
       // Show compilation progress bar and build time.
       new LogPlugin({

--- a/packages/docusaurus/src/webpack/client.ts
+++ b/packages/docusaurus/src/webpack/client.ts
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import chalk = require('chalk');
+import chalk from 'chalk';
 import path from 'path';
 import {Configuration} from 'webpack';
 import merge from 'webpack-merge';

